core/shared/communication/base-server-communication.ts

```diff
@@ -0,0 +1,46 @@
+import { Server, Socket } from 'net';
+import { Logger } from '../logger';
+import { BaseCommunication } from './base-communication';
+import { Message } from './message';
+
+export abstract class BaseServerCommunication extends BaseCommunication {
```

Comment:
What's the difference between `BaseServerCommunication` and `BaseSocketCommunication`?

Reply:
the one is the server, in our test case now cold-wallet => server and masternode manager => client / socket.
server is listening on a port for incoming connections
sockets is the one who is connecting to the servers' port